astroquery/exoplanet_archive/exoplanet_archive.py

```diff
@@ -0,0 +1,104 @@
+    def param_units(self):
+        if self._param_units is None:
+            module_dir = os.path.dirname(os.path.abspath(__file__))
+            units_file = open(os.path.join(module_dir, 'data',
+                                           'exoplanet_nexsci_units.json'))
+            self._param_units = json.load(units_file)
+
+        return self._param_units
+
+    def get_confirmed_planets_table(self, cache=True, show_progress=True):
+        """
+        Download (and optionally cache) the `NExScI Exoplanet Archive Confirmed
+        Planets table <http://exoplanetarchive.ipac.caltech.edu/index.html>`_.
+
+        The Exoplanet Archive table returns lots of columns of data. A full
+        description of the columns can be found `here
+        <https://exoplanetarchive.ipac.caltech.edu/docs/API_exoplanet_columns.html>`_
+
+        Parameters
+        ----------
+        cache : bool (optional)
+            Cache exoplanet table to local astropy cache? Default is `True`.
+        show_progress : bool (optional)
+            Show progress of exoplanet table download (if no cached copy is
+            available). Default is `True`.
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table of exoplanet properties.
+        """
+        if self._table is None:
+            table_path = download_file(EXOPLANETS_CSV_URL, cache=cache,
+                                       show_progress=show_progress, timeout=120)
+            exoplanets_table = ascii.read(table_path)
+
+            # Group by the host name
+            exoplanets_table.group_by('pl_hostname')
+
+            # Store column of lowercase names for indexing:
+            lowercase_names = [' '.join([host_name.lower(), letter])
+                               for host_name, letter in
+                               zip(exoplanets_table['pl_hostname'].data,
+                                   exoplanets_table['pl_letter'].data)]
+            exoplanets_table['NAME_LOWERCASE'] = lowercase_names
+            exoplanets_table.add_index('NAME_LOWERCASE')
+
+            # Assign units to columns where possible
+            for col in exoplanets_table.colnames:
+                if col in self.param_units:
+                    # Check that unit is implemented in this version of astropy
+                    if hasattr(u, self.param_units[col]):
+                        exoplanets_table[col].unit = u.Unit(self.param_units[col])
+
+            self._table = exoplanets_table
+
+        return self._table
+
+    def query_planet(self, planet_name):
+        """
+        Get table of exoplanet properties.
+
+        Parameters
+        ----------
+        planet_name : str
+            Name of planet
+
+        Return
+        ------
+        table : `~astropy.table.Table`
+            Table of one exoplanet's properties.
+        """
+
+        exoplanet_table = self.get_confirmed_planets_table()
+        return exoplanet_table.loc[planet_name.strip().lower()]
```

Comment:
Add `.replace(" ", "")` here to get rid of whitespaces, so that you can both support `query_planet("Trappist-1 b")` and `query_planet("Trappist-1b")`?  Right now the latter fails.